app/src/main/kotlin/com/wire/android/ui/authentication/create/common/CreateAccountBaseViewModel.kt

```diff
@@ -102,6 +105,23 @@ abstract class CreateAccountBaseViewModel(
     final override fun onEmailContinue() {
         emailState = emailState.copy(loading = true, continueEnabled = false)
         viewModelScope.launch {
+            fetchApiVersion(authServerConfigProvider.authServer.value).let {
+                when (it) {
+                    is FetchApiVersionResult.Success -> {}
+                    is FetchApiVersionResult.Failure.UnknownServerVersion -> {
+                        emailState = emailState.copy(showServerVersionNotSupportedDialog = true, loading = false)
+                        return@launch
+                    }
+                    is FetchApiVersionResult.Failure.TooNewVersion -> {
+                        emailState = emailState.copy(showClientUpdateDialog = true, loading = false)
+                        return@launch
+                    }
+                    is FetchApiVersionResult.Failure.Generic -> {
+                        return@launch
+                    }
+                }
+            }
```

Comment:
the screen will still be in the loading state in case of generic error 

you can use invokeOnCompletion to make sure the loading state is cleared
```suggestion
            fetchApiVersion(authServerConfigProvider.authServer.value).let {
                when (it) {
                    is FetchApiVersionResult.Success -> {}
                    is FetchApiVersionResult.Failure.UnknownServerVersion -> {
                        emailState = emailState.copy(showServerVersionNotSupportedDialog = true)
                        return@launch
                    }
                    is FetchApiVersionResult.Failure.TooNewVersion -> {
                        emailState = emailState.copy(showClientUpdateDialog = true)
                        return@launch
                    }
                    is FetchApiVersionResult.Failure.Generic -> {
                        return@launch
                    }
                }
            }.invokeOnCompletion { emailState = emailState.copy(loading = false) }
```